scripts/optimize_images/optimize-images.ts

```diff
@@ -0,0 +1,65 @@
+import fs from 'fs';
+import path from 'path';
+import sharp from 'sharp';
+import staticImages from './static-images';
+
+async function main() {
+  for (const imageInfo of staticImages) {
+    const { path: imagePath, ...sizeParams } = imageInfo;
+    const inputPath = path.join(__dirname, '../../', imagePath);
+
+    if (!fs.existsSync(inputPath)) {
+      console.log(`Image ${inputPath} not found, skipping.`);
+      continue;
+    }
+
+    // Rename the image `image.png` -> `image-original.png` (sharp doesn't allow
+    // the same input and output path).
+    const { dir: imageDir, name: fileName, ext } = path.parse(inputPath);
+    const backupName = `${fileName}-original${ext}`;
+    const backupImagePath = path.join(imageDir, backupName);
+
+    if (fs.existsSync(backupImagePath)) {
+      console.log(`Backup image ${backupName} found, skipping.`);
+      continue;
```

Comment:
I think instead of skipping the image entirely, we could maybe just skip the rename but still run sharp again.

That way if e.g. you need to resize an image, you can update the size in `static-images.ts` and just re-run `yarn optimize-images` and it'll do the right thing.  WDYT?

Reply:
Good point!